crates/oxc_linter/src/rules/nextjs/no_unwanted_polyfillio.rs

```diff
@@ -0,0 +1,289 @@
+    ///
+    /// <script src='https://polyfill.io/v3/polyfill.min.js?features=WeakSet%2CPromise%2CPromise.prototype.finally%2Ces2015%2Ces5%2Ces6'></script>
+    /// ```
+    NoUnwantedPolyfillio,
+    correctness
+);
+
+// Keep in sync with next.js polyfills file : https://github.com/vercel/next.js/blob/master/packages/next-polyfill-nomodule/src/index.js
+const NEXT_POLYFILLED_FEATURES: Set<&'static str> = phf_set! {
+    "Array.prototype.@@iterator",
+    "Array.prototype.at",
+    "Array.prototype.copyWithin",
+    "Array.prototype.fill",
+    "Array.prototype.find",
+    "Array.prototype.findIndex",
+    "Array.prototype.flatMap",
+    "Array.prototype.flat",
+    "Array.from",
+    "Array.prototype.includes",
+    "Array.of",
+    "Function.prototype.name",
+    "fetch",
+    "Map",
+    "Number.EPSILON",
+    "Number.Epsilon",
+    "Number.isFinite",
+    "Number.isNaN",
+    "Number.isInteger",
+    "Number.isSafeInteger",
+    "Number.MAX_SAFE_INTEGER",
+    "Number.MIN_SAFE_INTEGER",
+    "Number.parseFloat",
+    "Number.parseInt",
+    "Object.assign",
+    "Object.entries",
+    "Object.fromEntries",
+    "Object.getOwnPropertyDescriptor",
+    "Object.getOwnPropertyDescriptors",
+    "Object.is",
+    "Object.keys",
+    "Object.values",
+    "Reflect",
+    "Set",
+    "Symbol",
+    "Symbol.asyncIterator",
+    "String.prototype.codePointAt",
+    "String.prototype.endsWith",
+    "String.fromCodePoint",
+    "String.prototype.includes",
+    "String.prototype.@@iterator",
+    "String.prototype.padEnd",
+    "String.prototype.padStart",
+    "String.prototype.repeat",
+    "String.raw",
+    "String.prototype.startsWith",
+    "String.prototype.trimEnd",
+    "String.prototype.trimStart",
+    "URL",
+    "URL.prototype.toJSON",
+    "URLSearchParams",
+    "WeakMap",
+    "WeakSet",
+    "Promise",
+    "Promise.prototype.finally",
+    "es2015", // Should be covered by babel-preset-env instead.
+    "es2016", // contains polyfilled 'Array.prototype.includes', 'String.prototype.padEnd' and 'String.prototype.padStart'
+    "es2017", // contains polyfilled 'Object.entries', 'Object.getOwnPropertyDescriptors', 'Object.values', 'String.prototype.padEnd' and 'String.prototype.padStart'
+    "es2018", // contains polyfilled 'Promise.prototype.finally' and ''Symbol.asyncIterator'
+    "es2019", // Contains polyfilled 'Object.fromEntries' and polyfilled 'Array.prototype.flat', 'Array.prototype.flatMap', 'String.prototype.trimEnd' and 'String.prototype.trimStart'
+    "es5", // Should be covered by babel-preset-env instead.
+    "es6", // Should be covered by babel-preset-env instead.
+    "es7", // contains polyfilled 'Array.prototype.includes', 'String.prototype.padEnd' and 'String.prototype.padStart'
+};
+
+impl Rule for NoUnwantedPolyfillio {
+    fn run_once(&self, ctx: &LintContext<'_>) {
+        let mut script_import = "";
+        for node in ctx.nodes().iter() {
```

Comment:
We may not need this, can use take a look at https://github.com/oxc-project/oxc/blob/main/crates/oxc_syntax/src/module_record.rs and see if you can get the relevant information?

This is data is provided in `ctx.semantic().module_record()`

Reply:
replaced with `ctx.semantic().module_record()`, and also I replaced `run_once` with `run`. 
Let me know if I should use `run_once`